build.gradle.kts

```diff
@@ -24,7 +24,36 @@ dependencies {
     testImplementation(platform("org.junit:junit-bom:5.10.0"))
     testImplementation("org.junit.jupiter:junit-jupiter")
 }
+sourceSets {
+    val functionalTest by creating {
+        compileClasspath += sourceSets["main"].output + sourceSets["test"].output
+        runtimeClasspath += sourceSets["main"].output + sourceSets["test"].output
```

Comment:
we don't need classes compiled by tests to run functional tests, because we are not testing tests.